base/applications/cmdutils/more/more.c

```diff
@@ -140,28 +354,118 @@ PagePrompt(PCON_PAGER Pager, DWORD Done, DWORD Total)
     ConClearLine(Pager->Screen->Stream);
 
     /* Ctrl+C or Ctrl+Esc: Control Break */
-    if ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
-        ((KeyEvent.wVirtualKeyCode == L'C') &&
-         (KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED))))
+    if (fCtrl && ((KeyEvent.wVirtualKeyCode == VK_ESCAPE) ||
+                  (KeyEvent.wVirtualKeyCode == L'C')))
     {
         /* We break, output a newline */
         WCHAR ch = L'\n';
         ConStreamWrite(Pager->Screen->Stream, &ch, 1);
         return FALSE;
     }
 
-    /* 'Q': Quit */
-    // FIXME: Available only when command extensions are enabled.
-    if ((KeyEvent.wVirtualKeyCode == L'Q') &&
-        !(KeyEvent.dwControlKeyState & (LEFT_CTRL_PRESSED | RIGHT_CTRL_PRESSED)))
+    switch (chSubCommand)
     {
-        /* We break, output a newline */
-        WCHAR ch = L'\n';
-        ConStreamWrite(Pager->Screen->Stream, &ch, 1);
-        return FALSE;
+        case L'P':
+            chSubCommand = 0;
+            Pager->ScrollRows = nLines;
+            return TRUE;
+        case L'S':
+            chSubCommand = 0;
+            s_dwFlags |= FLAG_PLUSn;
+            s_nNextLineNo = Pager->lineno + nLines;
+            Pager->ScrollRows = Pager->ScreenRows - 1;
+            return TRUE;
+        default:
+            chSubCommand = 0;
+            break;
     }
 
-    return TRUE;
+    /* If extended features are available */
+    if (s_dwFlags & FLAG_E)
+    {
+        /* Ignore any key presses if Ctrl is pressed */
+        if (fCtrl)
+        {
+            chSubCommand = 0;
+            goto Restart;
+        }
+
+        /* 'Q' or [Esc]: Quit */
+        if (KeyEvent.wVirtualKeyCode == L'Q' ||
+            KeyEvent.wVirtualKeyCode == VK_ESCAPE)
+        {
+            /* We break, output a newline */
+            WCHAR ch = L'\n';
+            ConStreamWrite(Pager->Screen->Stream, &ch, 1);
+            return FALSE;
+        }
+
+        /* 'F': Next file */
+        if (KeyEvent.wVirtualKeyCode == L'F')
+        {
+            s_bDoNextFile = TRUE;
+            return FALSE;
+        }
+
+        /* '?': Show Options */
+        if (KeyEvent.uChar.UnicodeChar == L'?')
+        {
+            s_nPromptID = IDS_CONTINUE_OPTIONS;
+            goto Restart;
+        }
+
+        /* [Enter] key: Display one line */
+        if (KeyEvent.wVirtualKeyCode == VK_RETURN)
+        {
+            Pager->ScrollRows = 1;
+            return TRUE;
+        }
+
+        /* [Space] key: Display one page */
```

Comment:
Manual tests show that specific Space handling is done (with the support of the /C per-page screen clearing) only when extensions are enabled.